sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -1181,6 +1180,24 @@ generated precompiled preamble can be used:
     and `-Werror=pch-date-time`, which cause the compiler to diagnose a warning
     or error in this scenario.
 
+==== `--persistent-auto-pch=<dir>`
+
+Similar to `--auto-pch` but the precompiled preamble is stored on the filesystem
+at the location `<dir>`. That allows an application to precompile the preamble
+once and reuse it across multiple invocation of the application which isn't
+possible with `--auto-pch` that stores the precompiled preamble in-memory. This
+option is incompatible with `--auto-pch`.
+
+Some notes about the current behavior:
+
+* Threads only communicate via the persistent cache on the filesystem with no
+  other inter-thread communication to avoid extra synchronization overhead. In
+  an unlikely event that a preamble is missing from the cache and multiple
+  threads need it, each thread will build the preamble.
+
+* In the **very** unlikely event of a hash collision the precompiled preamble
+  from the persistent cache is ignored and the source is compiled without it.
```

Comment:
I assume this is the case when two different preambles happen to produce the same hash value?  Why is this specific to `--persistent-auto-pch`?  Wouldn't you have the same problem with `--auto-pch`?

Reply:
I can use full preamble+opts as the key in-memory:

https://github.com/intel/llvm/blob/93a515d0fb6bfe52595cef89da9a9e662c75c5c6/sycl-jit/jit-compiler/lib/rtc/DeviceCompilation.cpp#L92-L94

But I can't do that for filesystem storage (at least not without extra support like a list of different pch files sharing the hash). Another way to look at it is that `std::map<key, preamble>` handles the collisions for me without any extra efforts on my side.

Reply:
I see.  In that case, I would reword the note like:

> * Unlike `--auto-pch`, the `--persistent-auto-pch` option indexes the precompile preamble by a _hash_ of the preamble text, as opposed to the preamble text itself.  As a result, there is a highly unlikely possibility that two different preambles will produce the same hash value.  Even if this occurs, the compilation won't fail.  However, the preamble with the conflicting hash won't be cached, so subsequent compilations won't benefit from the caching.

Why is it difficult for the on-disk caching to use the actual text of the preamble as the key instead of computing a hash value?